src/collision.cpp

```diff
@@ -464,10 +467,12 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		bool step_up = false;
 		if (nearest_collided != COLLISION_AXIS_Y) {
 			aabb3f stepbox = movingbox;
-			stepbox.MinEdge.X += speed_f->X * dtime;
-			stepbox.MinEdge.Z += speed_f->Z * dtime;
-			stepbox.MaxEdge.X += speed_f->X * dtime;
-			stepbox.MaxEdge.Z += speed_f->Z * dtime;
+			// Look slightly ahead  for checking the height when stepping
```

Comment:
Why? What kind of issue does this fix? Please extend the comment.

Reply:
A precise computation is supposed to clip the X and Z values to just before the collision. But the `wouldCollideWithCeiling` function should check the ceiling also *after* stepping up, not assuming to stop before. Because the old computation used a too high speed, it would overestimate the position and hence test over the stair (usually, unless dtime is much too large).

Without this check, this might allow a "glitch elevator" for example by running against a stack of stairs.